src/connectors/bilibili.js

```diff
@@ -0,0 +1,46 @@
+import BaseConnector from 'content/base-connector';
+import Utils from 'content/utils';
+import _ from 'underscore';
+
+new class extends BaseConnector {
+    constructor() {
+        super();
+
+        this.name = 'Bilibili';
+        this.prefix = '/com/bilibili';
+
+        this.titleSelector = 'h1';
+
+        this.currentTimeSelector = '.bilibili-player-video-time-now';
+        this.lengthSelector = '.bilibili-player-video-time-total';
+
+        this.playButtonSelector =
+            '.bilibili-player-video-btn.bilibili-player-video-btn-start';
+        this.nextButtonSelector = '.bilibili-player-video-btn-next';
+
+        Utils.query('.player').then((elem) => this.observe(elem)
```

Comment:
This connector stops working when navigating to any other video from the panel on the right side. That's because that website replaces that `.player` element you start observing in constructor, so when video changes, you observe nothing. That could be solved with observing `div#bofqi`, which isn't replaced with new element in case of AJAX navigation. (I dunno if `bofqi` means something or is it just a random sequence of charachters, so I'd suggest to select that element with something like `div:has(> .player)`).

Reply:
It seems that Firefox can not recognize the selector `div:has(> .player)`, so I choose `div#bofqi`, and it worked.

BTW, `bofqi` means player in Chinese phonetic alphabets.